### DISCUSSION TITLE
Question: Support for MDX

### DISCUSSION THREAD

**Discussion by buremba** [ai/nanostores#256] [General]
I tried to use the library in MDX as follows:

```md
export const $counter = atom(10)

# Counter example

<div>
  <p>The counter currently has a value of {$counter.get()}.</p>
  <input value={$counter.get()} aria-label="Disabled slider"  onChange={(evt, newValue) => $counter.set(newValue)} />
</div>
```

While it correctly renders `10`, the changes are not propagated and the MDX renderer doesn't re-render the component again. Not sure if it's a silly question but is it feasible for nanostores to support MDX similar to how it supports React runtime?

**Comment by ai** [comment]
You need to find a way to call `useStore()` from `@nanostores/react` or `store.listen(rerenderComponent)` to subscribe to changes. Just calling `get()` just return current value.

I can’t help here since I do not use MDX.

**Reply by buremba** [reply]
`store.listen` works indeed but it requires me to embed some weird syntax in MDX, which is a dealbreaker for me now. I will try to see if I can modify MDX ast to rewrite it using `store.listen`. Thanks @ai !
